### ISSUE TITLE
Mode to ignore namespaces?

### ISSUE THREAD

**Issue by akx** [facelessuser/soupsieve#195]
Hey, it's me again with performance "issues" :)

My project needs to parse about, oh, 13,000 HTML files, and profiling the run and sorting by internal time gives me this top call list (unfortunately I don't have stats about the total duration of this run):

```
   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
    13441  120.725    0.009  610.285    0.045 bs4/builder/_lxml.py:320(feed)
 53003838   78.304    0.000  384.096    0.000 soupsieve/css_match.py:1287(match_selectors)
  9847474   64.891    0.000  327.462    0.000 bs4/builder/_lxml.py:213(start)
534596999   64.648    0.000   68.589    0.000 {built-in method builtins.isinstance}
 53310742   52.610    0.000  131.384    0.000 soupsieve/css_match.py:186(get_descendants)
 31676222   49.475    0.000  178.756    0.000 bs4/__init__.py:533(endData)
203441067   44.734    0.000   67.131    0.000 soupsieve/css_match.py:94(is_tag)
 55823816   43.801    0.000  116.995    0.000 soupsieve/css_match.py:628(match_namespace)
 45538107   42.938    0.000   42.938    0.000 bs4/element.py:158(setup)
 53003838   39.486    0.000  458.151    0.000 soupsieve/css_match.py:1402(match)
  9860915   37.393    0.000  110.214    0.000 bs4/element.py:1063(__init__)
217036857   36.962    0.000   37.652    0.000 bs4/element.py:1797(descendants)
  8315233   35.820    0.000   56.265    0.000 bs4/builder/__init__.py:284(_replace_cdata_list_attribute_values)
  9847474   33.851    0.000  242.638    0.000 bs4/__init__.py:664(handle_starttag)
 55823816   32.349    0.000   67.551    0.000 soupsieve/css_match.py:669(match_tagname)
 56003376   30.983    0.000  215.529    0.000 soupsieve/css_match.py:678(match_tag)
```

As you can see, 120 seconds is currently spent in `match_namespace`, and since I'm just scraping HTML 5 documents, I don't really care much about namespaces, and my selectors don't use them either.

Would it make sense to have a mode where namespace checks are entirely elided (at the user's own risk and so on)?

**Comment by facelessuser** [comment]
But they do use namespaces. HTML5 allows for, basically, implicit namespaces. That's how it kind of can replace xhtml and allow SVG etc. 

```py
>>> from bs4 import _soup
>>> soup = _soup('<div><svg></svg></div>', 'html5lib')
>>> soup.select_one('svg').namespace
'http://www.w3.org/2000/svg'
>>> soup.select_one('div').namespace
'http://www.w3.org/1999/xhtml'
```

**Comment by akx** [comment]
Sure, but what if I'd just want to pinky swear that I won't need namespace support at all? 
Come to think of it, maybe that could be on the user – since I'm in control of my pinky swears and dragon encounters, I can just do

```python
# Here be dragons
soupsieve.css_match._Match.match_namespace = lambda self, el, tag: True
```

in my program.

By the way, that story's different if you use lxml as the parser:

```python
>>> from bs4 import _soup
>>> soup = _soup('<div><svg></svg></div>', 'lxml')
>>> soup.select_one('svg').namespace
>>> soup.select_one('div').namespace
>>>
```

**Comment by facelessuser** [comment]
> since I'm in control of my pinky swears and dragon encounters, I can just do

I think this is your best bet, to be honest.

There are some pseudo classes that are only relevant in HTML, and I do use a shortcut to play off the namespaces to make that happen: https://github.com/facelessuser/soupsieve/blob/master/soupsieve/css_parser.py#L1162

> By the way, that story's different if you use lxml as the parser:

This is also true, but internally, if the document doesn't support namespaces, we assume the document is all XHTML: https://github.com/facelessuser/soupsieve/blob/master/soupsieve/css_match.py#L464. That feeds off of what I mentioned above by using namespaces to sandbox certain pseudo classes.

Basically, if the doc is XML or HTML5, namespaces exist, and if they aren't supported, we assume we are viewing an HTML document that doesn't support namespaces.

Anyways, it sounds like you have a solution that suits you, so I'm not sure there is anything for me to specifically do.

@gir-bot remove S: triage
@gir-bot add S: wontfix

**Comment by facelessuser** [comment]
I'm going to go ahead and close this. If I've have misinterpreted, and  you'd still like to discuss this more, feel free to reopen 🙂.
